kind/kind.mk

```diff
@@ -0,0 +1,43 @@
+kind_dir ?= .kind
+kind_marker := $(kind_dir)/.kind-setup_complete
+
+.PHONY: kind
+kind: export KUBECONFIG = $(KIND_KUBECONFIG)
+kind: kind-setup-ingress kind-load-image ## All-in-one kind target
+
+.PHONY: kind-setup
+kind-setup: export KUBECONFIG = $(KIND_KUBECONFIG)
+kind-setup: $(kind_marker) ## Creates the kind cluster
+
+.PHONY: kind-setup-ingress
+kind-setup-ingress: export KUBECONFIG = $(KIND_KUBECONFIG)
+kind-setup-ingress: kind-setup ## Install NGINX as ingress controller onto kind cluster
+	kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/main/deploy/static/provider/kind/deploy.yaml
+
+.PHONY: kind-load-image
+kind-load-image: kind-setup build-docker ## Load the container image onto kind cluster
+	@$(KIND) load docker-image --name $(KIND_CLUSTER) $(CONTAINER_IMG)
+
+.PHONY: kind-clean
+kind-clean: export KUBECONFIG = $(KIND_KUBECONFIG)
+kind-clean: ## Removes the kind Cluster
+	@$(KIND) delete cluster --name $(KIND_CLUSTER) || true
+	@rm -rf $(kind_dir)
+
+###
+### Artifacts
+###
+
+$(KIND_KUBECONFIG): export KUBECONFIG = $(KIND_KUBECONFIG)
+$(KIND_KUBECONFIG):
+	$(KIND) create cluster \
+		--name $(KIND_CLUSTER) \
+		--image $(KIND_IMAGE) \
+		--config kind/definitions/config.yaml
+	@kubectl version
+	@kubectl cluster-info
+
+$(kind_marker): export KUBECONFIG = $(KIND_KUBECONFIG)
+$(kind_marker): $(KIND_KUBECONFIG)
+	@kubectl config use-context kind-$(KIND_CLUSTER)
+	@touch $(kind_marker)
```

Comment:
Do we actually need this marker? Can't we use the $KIND_KUBECONFIG as a marker to see if the cluster exists?

Reply:
It is copy pasted from the K8up repo... not sure anymore why that is needed. Could be that it has to do with e2e testing, which we don't have here. I'll try it out.

Reply:
AFAICT, the marker file isn't strictly necessary. Let's remove it for now.